tools/system_libs.py

```diff
@@ -492,13 +497,15 @@ def build_objects(self, build_dir):
     commands = []
     objects = set()
     cflags = self.get_cflags()
+    source_dir = utils.path_from_root()
+    relative_source_dir = os.path.relpath(source_dir, build_dir)
     if self.deterministic_paths:
-      source_dir = utils.path_from_root()
-      if batch_inputs:
-        relative_source_dir = os.path.relpath(source_dir, build_dir)
-        cflags += [f'-ffile-prefix-map={relative_source_dir}/=']
-      cflags += [f'-ffile-prefix-map={source_dir}=/emsdk/emscripten',
-                 '-fdebug-compilation-dir=/emsdk/emscripten']
+      cflags += [f'-ffile-prefix-map={relative_source_dir}={FAKE_EMSCRIPTEN_PATH}']
```

Comment:
Can you split this line out into its own PR?   Something like "system_libs.py: Use the deterministic path when building in batch mode".    This seems like a separate and uncontroversial bugfix.

Reply:
Done: #23222

Reply:
Great!

Once #23222 lands maybe the next step should just be to unconditionally enable `determinisic_paths`?  i.e. just remove the non-deterministic mode.    We can always add it back later if it turns out someone wants it... but I doubt they wil.

Reply:
I thought what we were going to do was to unconditionally use `-fmacro-prefix-map` but use `-ffile-prefix-map` (which also enables `-fdebug-prefix-map`) only in `deterministic_paths`? This is basically what https://github.com/emscripten-core/emscripten/issues/23195#issuecomment-2549859260 suggested but you also said maybe even that was not necessary... But anyway this PR does not remove the undeterministic mode. I can do either.

Reply:
I rebased this PR on top of #23222 and #23225. This still uses `deterministic_paths`. Please let me know if you think it'd better go unconditionally set `deterministic_paths` and remove the variable, i.e., you don't need your real local paths in the debug info.

Reply:
I think just removing `deterministic_paths` and always doing it this way makes sense.  Sorry for the back on forth on this.